src/wallet/walletutil.cpp

```diff
@@ -25,3 +25,50 @@ fs::path GetWalletDir()
 
     return path;
 }
+
+static bool IsBerkeleyBtree(const fs::path& path)
+{
+    // Berkeley DB Btree magic bytes, from:
+    //  https://github.com/file/file/blob/5824af38469ec1ca9ac3ffd251e7afe9dc11e227/magic/Magdir/database#L74-L75
+    const uint32_t bdb_magic = 0x00053162;
+
+    if (!fs::exists(path)) return false;
+    if (fs::file_size(path) < 1024) return false;
```

Comment:
@ryanofsky @laanwj added these 2 checks to avoid opening unnecessary files.

Reply:
The first check `fs::exists(path)` was there to prevent exceptions from the 2nd check `fs::file_size(path)`. Fixed and added comment.